pages/signin.js

```diff
@@ -203,7 +203,12 @@ class SigninPage extends React.Component {
           }}
           submitRecoveryCode={recoveryCode => {
             const localStorage2FAToken = getFromLocalStorage(LOCAL_STORAGE_KEYS.ACCESS_TOKEN);
-            return this.props.login(localStorage2FAToken, { recoveryCode });
+            return this.props.login(localStorage2FAToken, { recoveryCode }).then(user => {
+              if (user?.collective?.slug) {
```

Comment:
That's still a bit weird as we're interrupting the login function which is supposed to return something. Maybe move to the `SignInOrJoinFree` `onSubmit` ?

Reply:
You will also be able to use the `router` instead of interacting directly with `window.location`.

Reply:
Makes sense, much better 👍🏻